File: Java/SauceDemo/src/main/java/pages/Sidebar.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class Sidebar {
    protected WebDriver driver;
    private final By react_burger_menu_btn = By.id("react-burger-menu-btn");
    private final By logout_btn = By.id("logout_sidebar_link");

    public Sidebar(WebDriver driver) {
        this.driver = driver;
    }

    public void clickReactBurgerMenuButton() {
        driver.findElement(react_burger_menu_btn).click();
    }

    public void clickLogoutButton() {
        driver.findElement(logout_btn).click();
    }
}

File: Python/src/pages/side_bar.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class Sidebar:
    """
    This class contains the elements of the sidebar
    """

    def __init__(self, driver):
        self.driver = driver

        # Locators
        self.react_burger_menu_btn = By.ID, "react-burger-menu-btn"
        self.logout_btn = By.ID, "logout_sidebar_link"

    def click_react_burger_menu_button(self):
        self.driver.find_element(*self.react_burger_menu_btn).click()

    def click_logout_button(self):
        # Esperar a que el botón de logout sea cliclable antes de hacer clic
        wait = WebDriverWait(self.driver, 10)
        logout_button = wait.until(EC.element_to_be_clickable(self.logout_btn))
        logout_button.click()
